scripts/linters/js_ts_linter.py

```diff
@@ -901,6 +901,98 @@ def _check_constants_declaration(self):
 
         return summary_messages
 
+    def _check_comments(self):
+        """This function ensures that comments follow correct style. Below are
+        some formats of correct comment style:
+        1. A comment can end with the following symbols: ('.', '?', ';', ',',
+        '{', '^', ')', '}', '>'). Example: // Is this is comment?
+        2. If a line contain any of the following words or phrases('@ts-ignore',
+        '--params', 'eslint-disable', 'eslint-enable', 'http://', 'https://')
+        in the comment.
+        """
+        if self.verbose_mode_enabled:
+            python_utils.PRINT(
+                'Starting comment checks\n'
+                '----------------------------------------')
+        summary_messages = []
+        files_to_check = self.all_filepaths
+        allowed_terminating_punctuations = [
+            '.', '?', ';', ',', '{', '^', ')', '}', '>']
+
+        # We are excluding individual comment line from this check if that
+        # line contain any of the below phrases anywhere in it. If we use
+        # punctuation at the end for the comment lines containing one of these
+        # phrases then it will mess up the whole meaning of the comment.
+        # Example: // eslint-disable-next-line max-len
+        # We can't use a period or any other symbol at the end of this line
+        # else eslint won't work as expected as this is a special comment for
+        # eslint.
+        allowed_comment_endings_without_period = [
+            '@ts-ignore', '--params', 'eslint-disable', 'eslint-enable',
+            'http://', 'https://']
+
+        failed = False
+        with linter_utils.redirect_stdout(sys.stdout):
+            for filepath in files_to_check:
+                file_content = FILE_CACHE.readlines(filepath)
+                file_length = len(file_content)
+                for line_num in python_utils.RANGE(file_length):
+                    line = file_content[line_num].strip()
+                    next_line = ''
+                    previous_line = ''
+                    if line_num + 1 < file_length:
+                        next_line = file_content[line_num + 1].strip()
+                    if line_num + 1 < file_length:
+                        next_line = file_content[line_num + 1].strip()
+
+                    # Check if comment contains any excluded phrase.
+                    comment_includes_allowed_comment_phrases = any(
+                        word in line for word in
+                        allowed_comment_endings_without_period)
```

Comment:
The variable name you have chosen does not match how you are handling it.

If you are just checking whether it is at the end of the comment, then presumably you would be using endswith() or similar.

Reply:
It is not guaranteed that these phrases always come at the end. That's why I didn't use endswith()

Reply:
Then why are you picking that variable name?

Make sure your variable names, your meaning, etc. are all consistent with each other.

Reply:
`allowed_phrases_in_the_comment` is it good?

Reply:
Per discussion, please make sure your variable names are precise, and that the checks catch exactly what they are supposed to catch, and do not catch what they are not supposed to catch.